### ISSUE TITLE
ReleasePipelineRunner does not trigger dependency updates when it commits a BuildChannel to the DB.

### ISSUE THREAD

**Issue by riarenas** [dotnet/arcade#2308]
The DB trigger that fires off the DependencyUpdater is scoped to the API.

This means that other services adding the Build to the channel won't fire off the dependency update.

**Comment by riarenas** [comment]
Testing this using int has proven a little painful. Right now I'm trying to investigate why Azure Devops is returning 400 when trying to clean up the artifact sources for a release pipeline.

That shouldn't be affected by this change, so I think it's just because the Int DB doesn't have all the right asset information the release needs, so I'm trying to push a real arcade build to the int DB (but that also requires to push an arcade-services build to int)

**Comment by jcagme** [comment]
We always deploy to arcade-services INT after every master build. Is that what you meant for the second part?

**Comment by riarenas** [comment]
No, I mean having the build, and all its dependencies in INT BAR

**Comment by JohnTortugo** [comment]
> Testing this using int has proven a little painful. Right now I'm trying to investigate why Azure Devops is returning 400 when trying to clean up the artifact sources for a release pipeline.

Is this a BadRequest error? If it is, usually the response from the server includes the reason for the error. Ping me on teams if you want to talk more about it.

**Comment by riarenas** [comment]
The error message is:

```
Artifact alias 'PrimaryArtifact' does not matching any linked artifact aliases for deploy job 'Agent job'. Specify a valid artifact alias and try again.
```
I was able to unblock this by manually removing the primary artifact source from the release pipeline and then the call (and every subsequent calls) are successful. It looks like this can be an inconsistent state that the release pipeline can get into, but I don't know yet. I'll open a new issue so we can understand this better and safeguard against it, but it's out of the scope of this issue.

**Comment by riarenas** [comment]
I'm satisfied with the validation of this issue now:
* Added build 209 to BAR-INT
* Default channel 131 got assigned to the build
* [This release](https://dev.azure.com/dnceng/internal/_releaseProgress?_a=release-pipeline-progress&releaseId=3861) was triggered
* Once the release finished, https://github.com/dotnet/arcade-validation/pull/216 was opened, triggered by an everyBuild dependency update.

**Comment by riarenas** [comment]
This has been rolled out, and this PR was opened by the everyBuild dependency update flow: https://github.com/dotnet/arcade-validation/pull/220
